srml/assets/src/lib.rs

```diff
@@ -22,6 +22,7 @@
 use srml_support::{StorageValue, StorageMap, Parameter, decl_module, decl_event, decl_storage, ensure};
 use primitives::traits::{Member, SimpleArithmetic, Zero, StaticLookup};
 use system::ensure_signed;
+use substrate_metadata_derive::EncodeMetadata;
```

Comment:
I tried this kind of code (as an alternative to the $crate in derive issue), seems ok but I did not test further, it might be a dead end or totally unrelated, just adding this comment in case it can help.
```diff
diff --git a/Cargo.lock b/Cargo.lock
index 63d7c26c..e0aaec65 100644
Binary files a/Cargo.lock and b/Cargo.lock differ
diff --git a/node/runtime/wasm/target/wasm32-unknown-unknown/release/node_runtime.compact.wasm b/node/runtime/wasm/target/wasm32-unknown-unknown/release/node_runtime.compact.wasm
index 33545160..2cf8a858 100644
Binary files a/node/runtime/wasm/target/wasm32-unknown-unknown/release/node_runtime.compact.wasm and b/node/runtime/wasm/target/wasm32-unknown-unknown/release/node_runtime.compact.wasm differ
diff --git a/srml/assets/Cargo.toml b/srml/assets/Cargo.toml
index 4d5d30b7..cca4c078 100644
--- a/srml/assets/Cargo.toml
+++ b/srml/assets/Cargo.toml
@@ -15,7 +15,7 @@ srml-support = { path = "../support", default-features = false }
 # `system` module provides us with all sorts of useful stuff and macros depend on it being around.
 system = { package = "srml-system", path = "../system", default-features = false }
 substrate-metadata = { path = "../../core/metadata", default-features = false }
-substrate-metadata-derive = { path = "../../core/metadata-derive", default-features = false }
+#substrate-metadata-derive = { path = "../../core/metadata-derive", default-features = false }
 
 [dev-dependencies]
 substrate-primitives = { path = "../../core/primitives", default-features = false }
@@ -31,5 +31,5 @@ std = [
 	"srml-support/std",
 	"system/std",
 	"substrate-metadata/std",
-	"substrate-metadata-derive/std",
+#	"substrate-metadata-derive/std",
 ]
diff --git a/srml/assets/src/lib.rs b/srml/assets/src/lib.rs
index 39014ad6..beda3d9c 100644
--- a/srml/assets/src/lib.rs
+++ b/srml/assets/src/lib.rs
@@ -22,7 +22,7 @@
 use srml_support::{StorageValue, StorageMap, Parameter, decl_module, decl_event, decl_storage, ensure};
 use primitives::traits::{Member, SimpleArithmetic, Zero, StaticLookup};
 use system::ensure_signed;
-use substrate_metadata_derive::EncodeMetadata;
+//use substrate_metadata_derive::EncodeMetadata;
 
 pub trait Trait: system::Trait {
 	/// The overarching event type.
diff --git a/srml/support/src/dispatch.rs b/srml/support/src/dispatch.rs
index b67a8959..143a0716 100644
--- a/srml/support/src/dispatch.rs
+++ b/srml/support/src/dispatch.rs
@@ -605,6 +605,9 @@ macro_rules! decl_module {
 			}
 		)*
 
+    mod dummy_import{
+      use super::*;
+	use $crate::substrate_metadata_derive::EncodeMetadata;
 		#[cfg(feature = "std")]
 		$(#[$attr])*
 		#[derive(EncodeMetadata)]
@@ -628,7 +631,8 @@ macro_rules! decl_module {
 				$fn_name ( $( $param ),* ),
 			)*
 		}
-
+    }
+    pub use dummy_import::*;
 		// manual implementation of clone/eq/partialeq because using derive erroneously requires
 		// clone/eq/partialeq from T.
 		impl<$trait_instance: $trait_name> $crate::dispatch::Clone
```

Reply:
thanks for the suggestion. this is definitely cleaner